apollo-compiler/src/test/graphql/com/example/simple_fragment/TestQuery.java

```diff
@@ -360,9 +360,7 @@ public int hashCode() {
 
       public static final class Mapper implements ResponseFieldMapper<Fragments> {
         static final ResponseField[] $responseFields = {
-          ResponseField.forFragment("__typename", "__typename", Arrays.<ResponseField.Condition>asList(
-            ResponseField.Condition.typeCondition(new String[] {"Human", "Droid"})
```

Comment:
Just double checking. Is this intentional? I don't really understand what's going on here and Im curious how this works.

Reply:
Sure, I will try to explain.

Let's imagine that we have this query:
```
query TestQuery {
  hero { # hero here has type Character
    ...HeroDetails
  }  
}
fragment HeroDetails on Character {
   name
   ...
}
```

Here, `hero` field has interface type `Character` with `Human`, `Droid` implementation in schema v1. 
Now if fragment `HeroDetails` is defined on interface type `Character` (not on concrete type) we were generating meta field like:
```
ResponseField.forFragment("__typename", "__typename", Arrays.<ResponseField.Condition>asList(ResponseField.Condition.typeCondition(new String[] {"Human", "Droid"})
```
with `Human` and `Droid` type conditions, means that in order to read fragment from the response it MUST be on of these types.

Now the issue that with a new version of schema that introduces new implementation of interface type `Character` let's say `Mandalorian` we are going to fail to read fragment (even when it's defined on base interface type `Character`) because we locked ourselves and expect only `Human` or `Droid` types.

This PR fixes this issue by generating this meta field instead:
```
ResponseField.forFragment("__typename", "__typename", Collections.<ResponseField.Condition>emptyList())
```
basically we remove type conditions if fragment defined on interface or union type.

Reply:
Thanks 👍